### PR TITLE
[RayService] Remove serve v1 API

### PR DESCRIPTION
**Pull request by architkulkarni**
This is the minimal PR to remove the Serve v1 API, which is removed in Ray 2.9.0.

- Deleted ServeDeploymentGraphSpec from the RayService CRD, along with its child data structures `RayActorOptionSpec` and `ServeConfigSpec`
- Ran `make manifests`
- Ran `make sync`

Followup work will include
- [ ] Remove the Serve v1 API from `apiserver` and `proto`.
- [ ] Delete `utils.SINGLE_APP` and `utils.MULTI_APP` and unify the codepaths that used them

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/rayservice_controller_test.go [hunk 1]
```diff
@@ -710,34 +672,14 @@ applications:
 		})
 
 		It("should reconcile status correctly when multi-app config type is used.", func() {
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayservice_controller_test.go:674]: Do we still need this test?

**Review by kevin85421 — COMMENTED** [review]
Could you double check `dashboard_httpclient.go` and `fake_serve_httpclient.go`? I think some functions and variables can also be deleted.

File: ray-operator/controllers/ray/rayservice_controller_test.go (hunk 1 above)
Comment (architkulkarni) [reply ray-operator/controllers/ray/rayservice_controller_test.go:674]: Removed it

**Comment by architkulkarni** [comment]
> Could you double check `dashboard_httpclient.go` and `fake_serve_httpclient.go`? I think some functions and variables can also be deleted.

@kevin85421 Yeah, would you prefer to do it all in this PR or is it okay to do it in a followup PR?  The current PR is the minimal PR that removes the v1 API.

**Comment by architkulkarni** [comment]
> > Could you double check `dashboard_httpclient.go` and `fake_serve_httpclient.go`? I think some functions and variables can also be deleted.
> 
> @kevin85421 Yeah, would you prefer to do it all in this PR or is it okay to do it in a followup PR? The current PR is the minimal PR that removes the v1 API.

I've done this in the following commit on a followup PR: https://github.com/ray-project/kuberay/pull/1786/commits/f439814792a1815c606dc08f7a13695f4fcac551 

We can merge this PR and then merge the followup PR, or if you prefer I can just add that commit to this PR.

**Review by kevin85421 — APPROVED** [review]

**Comment by kevin85421** [comment]
> would you prefer to do it all in this PR or is it okay to do it in a followup PR? The current PR is the minimal PR that removes the v1 API.

Both are fine for me. I like this PR. This PR makes the codebase more elegant!

**Pull request merged**